### PR TITLE
Localize TI dashboard

### LINKED ISSUE #5535: Localize the TI dashboard
**Is your feature request related to a problem? Please describe.**
It would be nice to make the TI dashboard more accessible to non-English speakers.

**Describe the solution you'd like**
We could offer translations for the text on the TI dashboard.

**Done when**
- All text on the TI Dashboard is translated into the languages that we support.

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Replace hard-coded messages with translated messages from i18n, to make the TI page accessible in other languages.

Note:
1. Some strings in current interface but not in the [new design](https://www.figma.com/file/yfhnUmE2FsonLtAqfeEfoo/TI-Edit?type=design&node-id=1-19&mode=design&t=5cHZQTZWZ0SiiMdX-0), these strings were not translated. For example, `Add client`  and `Email Adress`, the corresponding strings for new design are `Add new client` and `Email`.
2. `{0} applications submitted` in [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages#L159) needs to be modified, as we want the `s` in `applications` dynamic based on the number of applications.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)

### Issue(s) this completes

Partially fixes #5535

### CODE REVIEW

**Comment by azhangggggg** [comment]
> Thanks so much for doing this work! It seems really important for the trusted intermediaries.
> 
> I have a few suggestions:
> 
> * Would you mind adding one browser test just to make sure that translation is happening?  Maybe something similar to this: [translation test](https://github.com/civiform/civiform/blob/a27ea49b83fbbc4a10dffb7a96a8759974d49517/browser-test/src/trusted_intermediary.test.ts#L318)?
> * I think we'll need to localize the toast messages.  It may be as simple as using the `ToastMessage.error` method [here](https://github.com/civiform/civiform/blob/a27ea49b83fbbc4a10dffb7a96a8759974d49517/server/app/views/applicant/TrustedIntermediaryDashboardView.java#L118).
> * What’s the visual difference that’s causing all the snapshot changes?  I think we need to figure that out so that there isn't a small change that we didn't realize we were making.
> * We're also going to need to translate “Not yet logged in” and “Last login on“.  See this issue:  [Make statuses in the TI dashboard Accounts Settings table more informative #6255](https://github.com/civiform/civiform/issues/6255) .
> * "(no email address)" [here](https://github.com/civiform/civiform/blob/a27ea49b83fbbc4a10dffb7a96a8759974d49517/server/app/views/applicant/TrustedIntermediaryDashboardView.java#L422) may still be needed even though it isn't in the mockup.  Would you mind checking with @sijiayam ?

Thanks for the review.
1. A browser test added. Good call!
2. The toast messages, along with some other messages like `account status`, `Not yet logged in` will be translated in a separate PR. So will `(no email address)` once I verify it with Sijia. 
3. The difference between the snapshots are `All Clients`, it used to be `All clients`with the lowercase `c`. The uppercase `c`, which I believe aligns with the latest mock up. I will message Sijia to confirm.

File: server/app/views/applicant/TrustedIntermediaryDashboardView.java [hunk 1]
```diff
@@ -435,10 +451,12 @@ private TdTag renderStatusCell(AccountModel ti) {
...
   }
 
-  private TheadTag renderGroupTableHeader() {
+  private TheadTag renderGroupTableHeader(Messages messages) {
     return thead(
         tr().withClasses("border-b", "bg-gray-200", "text-left")
             .with(th("Info").withClasses(BaseStyles.TABLE_CELL_STYLES, "w-1/3"))
-            .with(th("Status").withClasses(BaseStyles.TABLE_CELL_STYLES, "w-1/4")));
+            .with(
+                th(messages.at(MessageKey.TITLE_STATUS.getKeyName()))
```
[the remark this answers is not in the document]
Comment (azhangggggg) [reply server/app/views/applicant/TrustedIntermediaryDashboardView.java]: I will add `Account Status` for translation in a separate PR, along with some other messages need to be translated.

File: server/app/views/applicant/TrustedIntermediaryDashboardView.java [hunk 2]
```diff
@@ -238,14 +249,14 @@ private DivTag renderAddNewForm(TrustedIntermediaryGroupModel tiGroup, Http.Requ
...
             .setValue(request.flash().get("providedDob").orElse(""));
     FieldWithLabel emailField =
         FieldWithLabel.email()
             .setId("email-input")
             .setFieldName("emailAddress")
-            .setLabelText("Email address" + OPTIONAL_INDICATOR)
+            .setLabelText("Email address " + messages.at(MessageKey.CONTENT_OPTIONAL.getKeyName()))
             .setToolTipIcon(Icons.INFO)
             .setToolTipText(
                 "Add an email address for your client to receive status updates about their"
```
[the remark this answers is not in the document]
Comment (azhangggggg) [reply server/app/views/applicant/TrustedIntermediaryDashboardView.java:262]: yeah no problem. I will verify it with Sijia.

File: server/app/views/applicant/TrustedIntermediaryDashboardView.java [hunk 3]
```diff
@@ -99,15 +99,18 @@ public Content render(
                 hr(),
                 renderSubHeader("Add Client").withId("add-client").withClass("my-4"),
                 requiredFieldsExplanationContent(),
-                renderAddNewForm(tiGroup, request),
+                renderAddNewForm(tiGroup, request, messages),
                 hr().withClasses("mt-6"),
-                renderSubHeader("All clients").withClass("my-4"),
+                renderSubHeader(messages.at(MessageKey.TITLE_ALL_CLIENTS.getKeyName()))
+                    .withClass("my-4"),
                 h4("Search"),
```
[the remark this answers is not in the document]
Comment (azhangggggg) [reply server/app/views/applicant/TrustedIntermediaryDashboardView.java:106]: Among the messages translated, we only have `Search` as `button.search`. I will add `Search` as part of the messages need to be translated in a separate PR, as `content.search`.

[the remark this answers is not in the document]
Comment (melanie-exygy) [reply server/app/views/applicant/TrustedIntermediaryDashboardView.java:106]: Okay, that makes sense.  What would you think of labelling the new message as `header.search` instead?

[the remark this answers is not in the document]
Comment (azhangggggg) [reply server/app/views/applicant/TrustedIntermediaryDashboardView.java:106]: sure.

**Review by melanie-exygy — APPROVED** [review]

**Pull request merged**